convert-hf-to-gguf.py

```diff
@@ -91,10 +131,27 @@ def __init__(self, dir_model: Path, ftype: gguf.LlamaFileType, fname_out: Path,
             else:
                 logger.info(f"choosing --outtype bf16 from first tensor type ({first_tensor.dtype})")
                 self.ftype = gguf.LlamaFileType.MOSTLY_BF16
-        ftype_up: str = self.ftype.name.partition("_")[2].upper()
-        ftype_lw: str = ftype_up.lower()
+
+        # Generate default filename based on model specification and available metadata
+        version_string = None # TODO: Add metadata support
+        expert_count = self.hparams["num_local_experts"] if "num_local_experts" in self.hparams else None
+        encodingScheme = {
+            gguf.LlamaFileType.ALL_F32     : "F32",
+            gguf.LlamaFileType.MOSTLY_F16  : "F16",
+            gguf.LlamaFileType.MOSTLY_BF16 : "BF16",
+            gguf.LlamaFileType.MOSTLY_Q8_0 : "Q8_0",
+        }[self.ftype]
```

Comment:
This is equivalent to

```python
self.ftype.name.partition("_")[2].upper()
```

but it's more verbose and doesn't support all existing types, so it's yet another place to update if new conversion types are added.